src/main/frontend/db_worker.cljs

```diff
@@ -251,31 +248,53 @@
 
   (getMaxTx
    [_this repo]
-   (when-let [conn (get-datascript-conn repo)]
+   (when-let [conn (state/get-datascript-conn repo)]
      (:max-tx @conn)))
 
   (q [_this repo inputs-str]
      "Datascript q"
-     (when-let [conn (get-datascript-conn repo)]
+     (when-let [conn (state/get-datascript-conn repo)]
        (let [inputs (edn/read-string inputs-str)]
          (let [result (apply d/q (first inputs) @conn (rest inputs))]
            (bean/->js result)))))
 
   (transact
-   [_this repo tx-data tx-meta]
-   (when-let [conn (get-datascript-conn repo)]
+   [_this repo tx-data tx-meta context]
+   (when repo (state/set-db-latest-tx-time! repo))
+   (when-let [conn (state/get-datascript-conn repo)]
      (try
-       (let [tx-data (edn/read-string tx-data)
-             tx-meta (edn/read-string tx-meta)
-             tx-report (d/transact! conn tx-data tx-meta)]
-         (search/sync-search-indice repo tx-report))
+       (let [tx-data (if (string? tx-data)
+                       (edn/read-string tx-data)
+                       tx-data)
+             tx-meta (if (string? tx-meta)
+                       (edn/read-string tx-meta)
+                       tx-meta)
+             context (if (string? context)
+                       (edn/read-string context)
+                       context)
+             _ (when context (state/set-context! context))
+             tx-meta' (if (:new-graph? tx-meta)
+                        tx-meta
+                        (-> tx-meta
+                            (assoc :skip-store? true) ; delay writes to the disk
+                            (dissoc :insert-blocks?)))]
+         (when-not (and (:create-today-journal? tx-meta)
```

Comment:
Avoid creating today's journal page repeatedly because it'll result in outliner broken chain, a journal page will have two empty block children that have the same parent (the journal page).